ci_framework/roles/edpm_deploy/tasks/main.yml

```diff
@@ -18,55 +18,74 @@
   vars:
-    make_edpm_deploy_env: "{{ cifmw_edpm_deploy_env }}"
-    make_edpm_deploy_dryrun: "{{ cifmw_edpm_deploy_dryrun | bool }}"
+    make_edpm_deploy_prep_env: "{{ cifmw_edpm_deploy_env }}"
+    make_edpm_deploy_prep_dryrun: "{{ cifmw_edpm_deploy_dryrun | bool }}"
   ansible.builtin.include_role:
     name: 'install_yamls_makes'
-    tasks_from: 'make_edpm_deploy'
+    tasks_from: 'make_edpm_deploy_prep'
 
-- name: Wait for EDPM compute deployment to finish
+- name: Kustomize and deploy OpenStackDataPlane
   when:
     - not cifmw_edpm_deploy_dryrun | bool
   environment:
     PATH: "{{ cifmw_path }}"
     KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
   block:
-    - name: Get info about dataplane node
+    - name: Find the OpenStack Dataplane CR manifest
+      ansible.builtin.find:
+        paths: "{{ cifmw_edpm_deploy_manifests_dir }}/{{ cifmw_install_yamls_defaults['NAMESPACE'] }}/dataplane/cr"
+        contains: "kind: OpenStackDataPlane"
+        excludes: "kind: Kustomization"
+        patterns: "*.yaml"
+      register: cifmw_edpm_deploy_cr_manifest_paths
+
+    - name: Ensure manifest exists
+      ansible.builtin.assert:
+        that: cifmw_edpm_deploy_cr_manifest_paths.matched == 1
+        quiet: true
+        msg: "Cannot determine OpenStackDataPlane deployment manifest"
+
+    - name: Perform kustomizations to the OpenStackDataPlane CR
+      vars:
+        cifmw_edpm_kustomize_cr_path: "{{ cifmw_edpm_deploy_cr_manifest_paths.files[0].path }}"
+        cifmw_edpm_kustomize_content: "{{ cifmw_edpm_deploy_dataplane_cr_kustomization | default('{}') }}"
+      ansible.builtin.include_role:
+        name: edpm_kustomize
+
+    - name: Apply the OpenStackDataPlane CR
+      environment:
+        KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+        PATH: "{{ cifmw_path }}"
+      when: not cifmw_edpm_deploy_dryrun
       ci_script:
         output_dir: "{{ cifmw_edpm_deploy_basedir }}/artifacts"
-        script: |-
-          oc get openstackdataplane -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }}
-          oc get openstackdataplanerole -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }}
-          oc get openstackdataplanenode -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }}
-          oc get pods -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }} | grep edpm
-      ignore_errors: true
-
-    - name: Get EDPM nova deployment pod name
-      register: edpm_pod
-      ansible.builtin.shell:
-        cmd: >-
-          oc get pods -o name
-          -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }} |
-          grep -m1  {{ cifmw_edpm_deploy_nova_pod_name }}
-      until: edpm_pod.rc == 0
-      retries: "{{ cifmw_edpm_deploy_retries }}"
-      delay: 50
+        script: "oc apply -f {{ cifmw_edpm_deploy_cr_manifest_paths.files[0].path }}"
 
-    - name: Wait for EDPM nova deployment pod to finish
-      register: deploy_status
+    - name: Wait for OpenStackDataplane to be deployed
```

Comment:
guess that's the replacement of the "wait for nova pod".

Reply:
Exactly, instead of waiting for an specific pod the idea now is to wait on the higher level resource to avoid situations like the one happening now, that is that the first deploy pod may fail and we don't realize that the next attempt succeed.